### ISSUE TITLE
Unable to start minikube v1.12.2 on Microsoft Windows 10 Enterprise 10.0.10586 Build 10586

### ISSUE THREAD

**Issue by aromalsbabu** [kubernetes/minikube#8977]
I would very much appreciate it if someone can help me with the following:

Windows Version: 10.0.10586
MiniKube Version: [v1.12.2]
[minikubeError.log](https://github.com/kubernetes/minikube/files/5059905/minikubeError.log)
[minikubeError.log](https://github.com/kubernetes/minikube/files/5059906/minikubeError.log)

Virtual Box Version: 6.1

Error:

Microsoft Windows [Version 10.0.10586]
(c) 2015 Microsoft Corporation. All rights reserved.

C:\Windows\system32>minikube start --driver=virtualbox
* minikube v1.12.2 on Microsoft Windows 10 Enterprise 10.0.10586 Build 10586
* Using the virtualbox driver based on user configuration
* Downloading VM boot image ...
    > minikube-v1.12.2.iso.sha256: 65 B / 65 B [-------------] 100.00% ? p/s 0s
    > minikube-v1.12.2.iso: 173.73 MiB / 173.73 MiB [ 100.00% 15.73 MiB p/s 11s
* Starting control plane node minikube in cluster minikube
* Downloading Kubernetes v1.18.3 preload ...
    > preloaded-images-k8s-v5-v1.18.3-docker-overlay2-amd64.tar.lz4: 510.91 MiB
* Creating virtualbox VM (CPUs=2, Memory=2200MB, Disk=20000MB) ...
* Deleting "minikube" in virtualbox ...
! StartHost failed, but will try again: creating host: create: provisioning: ssh command error:
command : sudo diff -u /lib/systemd/system/docker.service /lib/systemd/system/docker.service.new || { sudo mv /lib/systemd/system/docker.service.new /lib/systemd/system/docker.service; sudo systemctl -f daemon-reload && sudo systemctl -f enable docker && sudo systemctl -f restart docker; }
err     : Process exited with status 1
output  : diff: can't stat '/lib/systemd/system/docker.service': No such file or directory
Failed to enable unit: Unit file /usr/lib/systemd/system/docker.service is masked.

* Failed to start virtualbox VM. "minikube start" may fix it: Error loading existing host. Please try running [minikube delete], then run [minikube start] again.: filestore "minikube": open C:\Users\dmadmin\.minikube\machines\minikube\config.json: The system cannot find the file specified.
*
X error provisioning host: Failed to start host: Error loading existing host. Please try running [minikube delete], then run [minikube start] again.: filestore "minikube": open C:\Users\dmadmin\.minikube\machines\minikube\config.json: The system cannot find the file specified.
*
* minikube is exiting due to an error. If the above message is not useful, open an issue:
  - https://github.com/kubernetes/minikube/issues/new/choose

C:\Windows\system32>

I tried "minikube delete", followed by deleting folders ".minikube" and ".virtualbox" located under "C:\Users\<userid>". I then tried "minikube start --driver=virtualbox" but I still run into the same error. Any help would be appreciated
[minikubeError.log](https://github.com/kubernetes/minikube/files/5059911/minikubeError.log)
Thanks

**Comment by priyawadhwa** [comment]
Hey @aromalsbabu looks like minikube can't find the config for some reason:

```
open C:\Users\dmadmin.minikube\machines\minikube\config.json: The system cannot find the file specified.
```

it's a bit odd that even after deleting ~/.minikube this was still broken. 

I'm not sure if this will work, since manually deleting ~/.minikube didn't , but could you try:

```
minikube delete --all --purge # This should delete everything minikube related
minikube start
```

and let me know if that fixes your issue?

**Comment by aromalsbabu** [comment]
Thanks a lot, @priyawadhwa. Your comments did indeed help me figure out the problem. I cleaned up the minikube folder, started minikube again to see if config.json is getting generated. I noticed that it was getting generated and was getting deleted automatically (while minikube startup was failing). I took a copy of the config.json before it got deleted. From the  config.json, I got to know that NO_PORXY variable was set incorrectly on my VM. I corrected it and everything seems to be working ok. Thanks!
